client/analytics/report/revenue.js

```diff
@@ -212,13 +220,68 @@ class RevenueReport extends Component {
 		];
 	}
 
-	render() {
-		const { path, query } = this.props;
-		const { totals = {}, intervals = [] } = this.state.stats;
+	renderTable() {
+		const {
+			tableData,
+			isTableDataRequesting,
+			isTableDataError,
+			query,
+			previousTableData,
+		} = this.props;
+		const headers = this.getHeadersContent();
+
+		if ( isTableDataRequesting && isReportDataEmpty( previousTableData ) ) {
+			return (
+				<TableCardPlaceholder
+					rows={ 10 }
+					headers={ headers }
+					title={ __( 'Revenue Last Week', 'wc-admin' ) }
+				/>
+			);
+		}
+
+		// TODO Update Error Display
+		if ( isTableDataError ) {
+			return (
+				<div className="woocommerce-analytics__report-table-error">
+					{ __( 'There was an error retrieving your revenue summary. Please try again.' ) }
+				</div>
+			);
+		}
+
+		if ( isReportDataEmpty( tableData ) && ! isTableDataRequesting ) {
+			// TODO Return empty content component once it is built.
+			return null;
+		}
+
+		const data = isTableDataRequesting ? previousTableData : tableData;
+		const { totals = {}, intervals = [] } = ( data && data.data ) || {};
+		const { totalResults } = data;
 		const summary = this.getSummaryContent( totals ) || [];
 		const rows = this.getRowsContent( intervals ) || [];
-		const headers = this.getHeadersContent();
 
+		const className = classnames( 'woocommerce-analytics__report-table-card', {
+			'is-requesting': isTableDataRequesting,
+		} );
+
+		return (
+			<TableCard
+				title={ __( 'Revenue Last Week', 'wc-admin' ) }
+				rows={ rows }
+				totalRows={ totalResults }
+				headers={ headers }
+				onClickDownload={ noop }
+				onQueryChange={ this.onQueryChange }
+				query={ query }
+				summary={ summary }
+				className={ className }
+			/>
+		);
+	}
+
+	render() {
+		const { path, query } = this.props;
+		const { totals = {} } = this.state.stats;
```

Comment:
This `totals` is still fake data because we need to query for the table by week and _also_ query the previous week to get the deltas, right?

That kind of messes with my previous comment about how the table has been developed to _not_ use week queries, hm … 🤔

Reply:
Yes. I left this using fake data for now, because we will need different queries for the summary numbers/chart (including to get the delta).